client/engine/store/persiststore.go

```diff
@@ -0,0 +1,465 @@
+package store
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/statechannels/go-nitro/channel"
+	"github.com/statechannels/go-nitro/channel/consensus_channel"
+	"github.com/statechannels/go-nitro/crypto"
+	"github.com/statechannels/go-nitro/protocols"
+	"github.com/statechannels/go-nitro/protocols/directdefund"
+	"github.com/statechannels/go-nitro/protocols/directfund"
+	"github.com/statechannels/go-nitro/protocols/virtualdefund"
+	"github.com/statechannels/go-nitro/protocols/virtualfund"
+	"github.com/statechannels/go-nitro/types"
+	"github.com/tidwall/buntdb"
+)
+
+type PersistStore struct {
+	objectives         *buntdb.DB
+	channels           *buntdb.DB
+	consensusChannels  *buntdb.DB
+	channelToObjective *buntdb.DB
```

Comment:
A `buntdb.DB` corresponds to a key/value store and gets stored in a single file.

Reply:
Yup, each database can be configured seperately, using `db.SetConfig(config)`. This means we could have a different sync policy for each DB.